lib/Backend/Lower.cpp

```diff
@@ -2896,7 +2896,23 @@ Lowerer::LowerRange(IR::Instr *instrStart, IR::Instr *instrEnd, bool defaultDoFa
         case Js::OpCode::StatementBoundary:
             // This instruction is merely to help convey source info through the IR
             // and eventually generate the nativeOffset maps.
-            break;
+#if DBG_DUMP && DBG
+            // If we have a JITStatementBreakpoint, then we should break on this statement
```

Comment:
It's not necessarily an either/or situation; this functionality should be useful on its own, at least, for the reasons you mentioned earlier (work with unmodified JS code), and additionally that it should allow you to keep much more of the pipeline in a steady state (since a debugbreak instruction might have different semantics or effects on earlier stages in the pipeline, at the very least by changing parse structures).